src/sidebar/services/api.js

```diff
@@ -128,18 +140,29 @@ function serializeParams(params) {
  * @param links - Object or promise for an object mapping named API routes to
  *                URL templates and methods
  * @param route - The dotted path of the named API route (eg. `annotation.create`)
- * @param {Function} tokenGetter - Function which returns a Promise for an
- *                   access token for the API.
+ * @param [APIMethodOptions] - Configuration for the API method
  * @return {APICallFunction}
  */
-function createAPICall($http, $q, links, route, tokenGetter) {
+function createAPICall(
+  $http,
+  $q,
+  links,
+  route,
+  {
+    getAccessToken = noop,
+    onRequestStarted = noop,
+    onRequestFinished = noop,
+  } = {}
+) {
   return function(params, data, options = {}) {
+    onRequestStarted();
+
     // `$q.all` is used here rather than `Promise.all` because testing code that
     // mixes native Promises with the `$q` promises returned by `$http`
     // functions gets awkward in tests.
     let accessToken;
     return $q
-      .all([links, tokenGetter()])
+      .all([links, getAccessToken()])
       .then(([links, token]) => {
```

Comment:
You mean "Promise" rather than "fn reference" I presume. `links` is the result of the `apiRoutes.links()` method. Perhaps it is slightly confusing to have variables of different types (Promise, concrete object) but the same name within an enclosing function. It gets a bit annoying to have to invent different names for the result though.